.github/workflows/pdk-basic.yml

```diff
@@ -0,0 +1,52 @@
+# This is a basic workflow to help you get started with Actions
+
+name: PDK basic
+
+# Controls when the workflow will run
+on:
+  workflow_call:
+    inputs:
+      container_image:
+        description: Image to use when running tests
+        default: 'puppet/puppet-dev-tools:2022-03-28-92c7176'
+        required: false
+        type: string
+
+
+env:
+  GITHUB_USER: ${{ secrets.PUPPET_REPO_USER }}
+  GITHUB_TOKEN: ${{ secrets.PUPPET_REPO_USER_TOKEN }}
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  validate:
```

Comment:
In our other tests we only run validations once. Mostly because I don't recall seeing a difference in static validations between versions. Perhaps when the Puppet language was still evolving this made sense, but now I think it's not worth the additional compute. Also, you may end up with multiple annotations for the same line. For example, both puppet-lint and rubocop can do that now (see https://github.com/puppetlabs/puppet-lint/pull/34, though not supported for PDK now).

Reply:
Yeah, I would agree here.

Reply:
This no longer needs to be a matrix now.